lib/galaxy/tools/__init__.py

```diff
@@ -2185,6 +2184,13 @@ def handle_input(
             collection_info=collection_info,
             completed_jobs=completed_jobs,
         )
+
+        tags = incoming.get("tags", [])
```

Comment:
You'll also want to assert that you're dealing with a list of strings here and raise an appropriate RequestParameterInvalidException if that's not the case.